core/includes/actions.inc

```diff
@@ -73,16 +73,16 @@ function actions_get_info($action_name = NULL) {
  *
  * @param string $action_name
  *   The machine name of the action to be executed.
- * @param Entity $entity
+ * @param ?Entity|null $entity
  *   The entity object that the action will act upon, such as a node, user, or
  *   comment object.
  *
- * @return
+ * @return mixed
  *   The result of the executed action callback.
  *
  * @ingroup actions
  */
-function actions_execute($action_name, Entity $entity = NULL, &$context) {
+function actions_execute($action_name, ?Entity $entity = NULL, &$context) {
```

Comment:
Optional parameter $entity declared before required parameter $context (on php 8.3).

Reply:
Hm, sure enough. Well that means if $context is required, Entity doesn't need to be optional (but it still needs to be nullable). So this:

```suggestion
function actions_execute($action_name, ?Entity $entity, &$context) {
```

Reply:
You seem to have fixed it "in mind", but forgot to actually fix it in code. At least, you didn't commit/push it.

The nagging persists, as it's still `?Entity $entity = NULL`. (Unfortunately I can't "unresolve" here permission-wise.)

Reply:
I removed the NULL.